iptables/iptables_test.go

```diff
@@ -0,0 +1,28 @@
+package iptables
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestMakeMultiportDestinations(t *testing.T) {
+	assertEqual(t, makeMultiportDestinations([]string{}), [][]string{})
+	assertEqual(t, makeMultiportDestinations([]string{"22", "25-27", "33"}), [][]string{{"22", "25:27", "33"}})
+	assertEqual(t, makeMultiportDestinations([]string{"22", "25-27", "not-a-number", "33"}), [][]string{{"22", "25:27", "33"}})
+	assertEqual(t, makeMultiportDestinations([]string{"22", "25-27", "notanumber", "33"}), [][]string{{"22", "25:27", "33"}})
+}
+
+func TestMakeMultiportDestinations_Split(t *testing.T) {
+	assertEqual(t,
+		makeMultiportDestinations([]string{"22-23", "25-27", "33-34", "35-35", "37-38", "50-54", "56-57", "60-63"}),
+		[][]string{{"22:23", "25:27", "33:34", "35:35", "37:38", "50:54", "56:57"}, {"60:63"}})
+	assertEqual(t,
+		makeMultiportDestinations([]string{"22-23", "25-27", "33-34", "35-35", "37-38", "50-54", "56", "58", "60", "63", "70-72"}),
+		[][]string{{"22:23", "25:27", "33:34", "35:35", "37:38", "50:54", "56", "58", "60"}, {"63", "70:72"}})
```

Comment:
I wonder, would it make sense to rewrite port ranges that are with equal upper and lowe bounds as a single port. Like for example 35-35 to just 35 rather than 35:35 ?

Reply:
This has now been updated to "compress" single-port port ranges.